### PR TITLE
feat(ci): A new workflow to validate TF code against different TF versions

### PR DESCRIPTION
**Pull request by FoSix**
## Description

A manually triggered workflow (for the moment) that runs `terraform validate` against all modules and examples. It does it for different TF versions. Currently these are: 0.15.0, 1.0.0, 1.2.0.

Code is run in threads, meaning that each module/example is tested against all TF versions at the same time. 

Since testing required commits to `develop` branch, development was done on a [fork repo, `develop` branch](https://github.com/FoSix/terraform-azurerm-vmseries-modules/actions/runs/2846100286). This PR is only the final code. 

## Motivation and Context

We need to make sure that while developing the code with the latest TF versions the code stays compatible with the declared range. Doing this manually is almost impossible. 

## How Has This Been Tested?

Workflow was run locally using `act` and on a forked repo (a commit to develop branch was required)

## Screenshots (if appropriate)

Validate_examples_and_modules_against_variety_of_TF_versions_·_FoSix_terraform-azurerm-vmseries-modules_e6b6b05_-_Vivaldi

## Types of changes

- New feature (non-breaking change which adds functionality)

## Checklist

- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes if appropriate.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: .github/workflows/tf_validate_ver.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
+---
+name: Validate examples and modules against variety of TF versions
+
+on:
+  # push:
+  #   branch:
+  #     - develop
```
Comment (pimielowski) [inline .github/workflows/tf_validate_ver.yml:7]: Do we enable it?

**Review by pimielowski — APPROVED** [review]
Looks good! I test it locally, it work fine :)

File: .github/workflows/tf_validate_ver.yml (hunk 1 above)
Comment (FoSix) [reply .github/workflows/tf_validate_ver.yml]: at some point - yes, initially the idea is to run it manually and then merge it into the 'regular' ci/cd workflow

File: .github/workflows/tf_validate_ver.yml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  workflow_dispatch:
+
+env:
+  # tf_versions needs to be a string of TF versions we would like to test against
+  # versions have to be space delimited
+  tf_versions: 0.15.0 1.0.0 1.2.0
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/tf_validate_ver.yml:10]: the line is configurable and I set it up as an example, so let's have a bigger discussion how to set it up. We can step over each minor version, but then the output will be huge (for these 3 this is almost 50 steps to look at). On the other hand it does not hurt, we can do a very detailed check with this workflow.

File: .github/workflows/tf_validate_ver.yml [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        tf_versions: ${{ fromJson(needs.prerequisites.outputs.tf_versions) }}
+        modules: ${{ fromJson(needs.prerequisites.outputs.modules) }}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+      - name: setup Terraform
+        uses: hashicorp/setup-terraform@v2
+        with:
+          terraform_version: ${{ matrix.tf_versions }}
+      - name: run modules validation
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/tf_validate_ver.yml]: I've changed the name to be shorter and more descriptive, and this particular name will not get displayed anymore as it is a part of a local action

File: .github/workflows/tf_validate_ver.yml [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      examples: ${{ steps.preqs.outputs.examples }}
+      tf_versions: ${{ steps.preqs.outputs.tf_versions }}
+    steps:
+      - name: checkout code
+        uses: actions/checkout@v3
+      - name: set outputs
+        id: preqs
+        run: |
+          echo "::set-output name=modules::$(find modules -type d -not \( -name ".?*" \) -maxdepth 1 -mindepth 1 | jq -R -s -c 'split("\n")[:-1]')"
+          echo "::set-output name=examples::$(find examples -type d -not \( -name ".?*" \) -maxdepth 1 -mindepth 1 | jq -R -s -c 'split("\n")[:-1]')"
```
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/tf_validate_ver.yml]: fixed

File: .github/workflows/tf_validate_ver.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (FoSix) [reply .github/workflows/tf_validate_ver.yml]: so finally we do `tf_versions: 0.15 1.0 1.1 1.2` - this brings in the latest patch level from each minor version.

**Pull request merged**